tasks/api/src/main/java/org/projectnessie/nessie/tasks/api/TaskRequest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.nessie.tasks.api;
+
+import java.util.concurrent.CompletionStage;
+import org.projectnessie.versioned.storage.common.persist.ObjId;
+import org.projectnessie.versioned.storage.common.persist.ObjType;
+
+/** Base interface for value objects that identify a task's input parameters. */
+public interface TaskRequest<T extends TaskObj, B extends TaskObj.Builder> {
```

Comment:
`T` is not used. Also, `B` should be a builder for `T`, but it is not apparent. 

How about this?
`TaskRequest<T extends TaskObj, B extends TaskObj.Builder<T>>` and
`interface Builder<T extends TaskObj>` and
```
Builder<T> from(TaskObj base);
...
T build();
```